libdevcore/CommonData.h

```diff
@@ -183,6 +183,12 @@ template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U con
 		_a.push_back(i);
 	return _a;
 }
+/// Concatenate the contents of a container onto a vector, move variant.
+template <class T, class U> std::vector<T>& operator+=(std::vector<T>& _a, U&& _b)
+{
+	std::copy(std::make_move_iterator(_b.begin()), std::make_move_iterator(_b.end()), std::back_inserter(_a));
```

Comment:
Don't you want `std::move` instead of `std::copy` here? http://en.cppreference.com/w/cpp/algorithm/move

Reply:
It should be identical, if you use `make_move_iterator`, but perhaps a direct move is better, yes.